### ISSUE TITLE
能否对openai的回复也进行敏感词过滤？

### ISSUE THREAD

**Issue by Cliffe97** [zhayujie/chatgpt-on-wechat#816]
### 前置确认

1. 网络能够访问openai接口
2. python 已安装：版本在 3.7 ~ 3.10 之间
3. `git pull` 拉取最新代码
4. 执行`pip3 install -r requirements.txt`，检查依赖是否满足
5. 拓展功能请执行`pip3 install -r requirements-optional.txt`，检查依赖是否满足
6. 在已有 issue 中未搜索到类似问题
7. [FAQS](https://github.com/zhayujie/chatgpt-on-wechat/wiki/FAQs) 中无类似问题

### 问题描述

> 简要说明、截图、复现步骤等，也可以是需求或想法

现在的banwords插件可以对用户提问进行审查，但是openai的回复还是无限制的。

### 终端日志 (如有报错)

```
[在此处粘贴终端日志, 可在主目录下`run.log`文件中找到]
```

### 环境

 - 操作系统类型  (Mac/Windows/Linux)：
 - Python版本  ( 执行 `python3 -V` )：                      
 - pip版本  ( 依赖问题此项必填，执行 `pip3 -V`)：

**Comment by lanvent** [comment]
openai回答有自己的内容政策限制吧，可以举个例子吗

**Comment by Cliffe97** [comment]
@lanvent ,比如说xijinping这三个字，如果加入banword，那用户是无法提出带这三字的问题的。但是如果问，中国国家主席是谁，那chatgpt api还是会回答。如果能对回答也进行过滤，那可以规避很多法律风险。

**Comment by goldfishh** [comment]
openai提供对输出文本内容的分类：moderation, 可以用这个接口返回值过滤定性的内容，如果你想在openai 接口层特殊定制
可以试一试：
logit_bias map Optional Defaults to null
Modify the likelihood of specified tokens appearing in the completion.

Accepts a json object that maps tokens (specified by their token ID in the tokenizer) to an associated bias value from -100 to 100. Mathematically, the bias is added to the logits generated by the model prior to sampling. The exact effect will vary per model, but values between -1 and 1 should decrease or increase likelihood of selection; values like -100 or 100 should result in a ban or exclusive selection of the relevant token.

openai api这个字段控制生成token采样概率，理论上说维护一个map传入，这些token不会被采样

**Comment by lanvent** [comment]
改了一点，现在可以开启回复过滤了。

**Comment by Cliffe97** [comment]
谢谢！

**Comment by Black-pupil** [comment]
我在部署项目后遇到了这个敏感词插件失效的问题，python3.9
  ![1685009924614](https://github.com/zhayujie/chatgpt-on-wechat/assets/65754817/0feec714-db5e-4eca-822f-146ec1de128d)
除了config.json和banwords.txt外没有做任何修改。
  ![1685009954393](https://github.com/zhayujie/chatgpt-on-wechat/assets/65754817/6086cdf9-7260-4a86-8f79-cd6f65dd392d)

**Comment by dr-data** [comment]
lm
